### PR TITLE
20915 add logging to x reference requests

### PR DESCRIPTION
**Pull request by RoelWKramer**
Log request id to cross reference requests to gob api.

* Fix secure_user errors
* Add flask mock app fixture

### CODE REVIEW

File: src/gobexport/worker.py [hunk 1]
```diff
@@ -18,51 +24,48 @@ class Worker():
...
+                response.raise_for_status()
 
-                # yield result
-                for line in result.iter_lines():
+                for line in response.iter_lines():
                     yield line
             except Exception as e:
-                print(f"Worker result {worker_id} failed")
-                # Re-raise exception
+                logger.info(f"Worker result {worker_id} failed")
```
Comment (RoelWKramer) [inline src/gobexport/worker.py:64]:
```suggestion
                logger.error(f"Worker result {worker_id} failed", exc_info=True)
```

File: src/gobexport/app.py [hunk 2]
```diff
@@ -168,14 +167,14 @@ def dump_on_new_events(msg):
 def run_message_thread():
     try:
         messagedriven_service(SERVICEDEFINITION, "Export")
-    except:  # noqa: E722 do not use bare 'except'
-        pass
-    print("ERROR: no connection with GOB message broker, application is stopped")
-    os._exit(os.EX_UNAVAILABLE)
+    except Exception as e:  # noqa: E722 do not use bare 'except'
```
Comment (fzonneveld-adam) [inline src/gobexport/app.py:170]: Comment can go

File: src/gobexport/app.py [hunk 3]
```diff
@@ -168,14 +167,14 @@ def dump_on_new_events(msg):
...
+        print("ERROR: no connection with GOB message broker, application is stopped")
+        print(e)
+        os._exit(os.EX_UNAVAILABLE)
```
Comment (fzonneveld-adam) [inline src/gobexport/app.py:173]: Original code is always exits with os.exit, not only on exception.  Intentional change?

File: src/gobexport/__main__.py [hunk 4]
```diff
@@ -1,9 +1,5 @@
...
 
-
-def init():
-    if __name__ == "__main__":
-        run_app()
-
-
-init()
+if __name__ == "__main__":
+    # pragma: no cover
```
Comment (fzonneveld-adam) [inline src/gobexport/__main__.py:4]: # pragma: no cover should be after if __name__ == clause to exclude everything in if clause from coverage

File: src/gobexport/flask_api.py [hunk 5]
```diff
@@ -55,20 +58,23 @@ def _products():
     return result
 
 
-def get_flask_app():
+def get_flask_app(config: Optional[Dict[str, any]] = None):
```
Comment (fzonneveld-adam) [inline src/gobexport/flask_api.py:61]: return -> flask.Flask

File: src/gobexport/keycloak.py [hunk 6]
```diff
@@ -26,9 +28,7 @@ def _init_credential_store(secure_user):
 def get_secure_header(secure_user):
     """
     Get the request header to access secure endpoints
-
     """
-    assert secure_user, "A secure_user must be defined to request a secure url"
```
Comment (fzonneveld-adam) [inline src/gobexport/keycloak.py:31]: Removed?

File: src/gobexport/requests.py [hunk 7]
```diff
@@ -52,28 +57,26 @@ def _exec(method, url, headers, **kwargs):
...
             print(f"Retry in {_RETRY_TIMEOUT} seconds, retries left: {_MAX_TRIES - n_tries}")
             time.sleep(_RETRY_TIMEOUT)
-            # Update headers because access token might be expired
-            headers = _updated_headers(url=url, headers=headers)
 
     request = ', '.join([f"{k}='{v}'" for k, v in kwargs.items()])
     raise APIException(f"Request '{request}' failed, tried {n_tries} times")
 
 
 def _updated_headers(url, headers=None, secure_user=None):
```
Comment (fzonneveld-adam) [inline src/gobexport/requests.py:65]: Why secure_user = None, will raise assertion.

File: src/tests/conftest.py [hunk 8]
```diff
@@ -0,0 +1,20 @@
+from collections import Generator
+
+import pytest
+from flask import Flask
+
+from gobexport.flask_api import get_flask_app
+
+
+@pytest.fixture
+def app() -> Generator[Flask, None, None]:
```
Comment (fzonneveld-adam) [inline src/tests/conftest.py:10]: Very nice 👍

File: src/gobexport/app.py (hunk 3 above)
Comment (RoelWKramer) [reply src/gobexport/app.py:173]: nice catch, not tested though ;-) 100% coverage ftw!

File: src/gobexport/requests.py (hunk 7 above)
Comment (RoelWKramer) [reply src/gobexport/requests.py:65]: Probably a nice idea to elaborate that in the docstring.

File: src/gobexport/requests.py [hunk 9]
```diff
@@ -24,7 +28,7 @@ class APIException(IOError):
     pass
 
 
-def _exec(method, url, headers, **kwargs):
+def _exec(method, url, secure_user: Optional[str] = None, **kwargs):
```
Comment (RoelWKramer) [inline src/gobexport/requests.py:31]: self review: complete type information here

File: src/gobexport/requests.py [hunk 10]
```diff
@@ -100,30 +103,30 @@ def wrapper(*args, **kwargs):
...
-        result = Worker.handle_response(result)
-    except requests.exceptions.RequestException:
-        raise APIException(f"Request failed due to API exception, response code {result and result.status_code}")
-    return result
+        response = requests.post(
+            url, headers=_updated_headers(url, Worker.headers, secure_user), stream=True, json=json, **kwargs)
+        response.raise_for_status()
+        return Worker.handle_response(response)
+    except requests.exceptions.RequestException as e:
+        # add request id here
```
Comment (RoelWKramer) [inline src/gobexport/requests.py:126]: comment can go

File: src/gobexport/dump.py [hunk 11]
```diff
@@ -155,17 +155,17 @@ def try_dump_collection(self, schema, catalog_name, collection_name, force_full=
                 stream=True
             )
 
-            lastline = ""
+            last_line = ""
             start_line = time.time()
-            for line in result.iter_lines(chunk_size=1):
-                lastline = line.decode()
+            for line in result.iter_lines(chsunk_size=1):
```
Comment (roel-devries) [inline src/gobexport/dump.py:160]:
```suggestion
            for line in result.iter_lines(chunk_size=1):
```

**Review by roel-devries — APPROVED** [review]

**Pull request merged**